### ISSUE TITLE
[Marketplace] Routing issues with multiple shops

### ISSUE THREAD

**Issue by mikemurray** [reactioncommerce/reaction#2760]
### Expected behavior
- Router should be able to resolve routes for the primary shops as well as all marketplace shops using `Router.go` and `Router.pathFor`.
- Should be able to navigate to a marketplace shop's homepage. For example `/reaction1` and view its products.

### Actual Behavior
- The router can only resolve the current stop and the primary shop routes.
- There is no index route for shops. for example, if your shop URL was `/reacton1`, you would get a 404 page because that route doesn't exist

### Steps to Reproduce the Behavior
1) Add a new marketplace shop in settings
2) Go to slug for shop in address bar, see the 404

### Versions
Node: 7.2.0
NPM: 5.3.0
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.5.0
Reaction branch: marketplace
Docker: 17.06.1-ce

**Comment by spencern** [comment]
It might be worth considering passing in either a shopId or shop slug into `Router.go` and `Router.pathFor`

**Comment by spencern** [comment]
@mikemurray are you working to resolve this? Do you think it's a bug or an extension of existing features?

**Comment by mikemurray** [comment]
@spencern Well, currently you get a 404 if you go to any presumed shop route, such as `/reaction1`, when you should get the index page for that shop. So Bug?

We need to add routes like the following options.

- `/:shopHandle` or `/shops/:shopHandle` & `/shops` for all shops?
- `/:shopHandle/tag/:id?` or `/shops/:shopHandle/tag/:id?`
- `/:shopHandle/product/:handle` or `/shops/product/shopHandle/:handle`

I am not currently working to resolve this issue.

**Comment by spencern** [comment]
@mikemurray It does sound like a bug then.
Is this in your queue or is anyone else competent with your router to add these options?

Are you blocked by this right now?

**Comment by mikemurray** [comment]
It is not in my queue and was punted to @jshimko to fix the router issue as he may have some ideas. As far as I know only you, @zenweasel and myself have worked on the router.

This issue is blocking progress in terms of properly testing filtering products by the shop on the grid in both marketplace and elsewhere.

**Comment by aaronjudd** [comment]
@jshimko this is something that is a blocker, and we need to know if you're going to be able to work on this, this week?

**Comment by spencern** [comment]
Taking this issue today, @mikemurray and I are pairing on it. check the `spencer-2760-marketplace-routing` branch to follow.

**Comment by mikemurray** [comment]
This is fixed with the merging of #2929, which addresses routing issues for marketplace shops.

@spencern While some other concerns remain, like actually putting links to shops in the navigation bar, I think this works as intended.

**Comment by spencern** [comment]
Closing as #2929 has been merged.

If there is need to add a navigation item for browsing shops, we should open another issue for that.
